metron-platform/metron-writer/src/main/java/org/apache/metron/writer/hdfs/HdfsWriter.java

```diff
@@ -82,38 +90,48 @@ public void init(Map stormConfig, TopologyContext topologyContext, WriterConfigu
     this.fileNameFormat.prepare(stormConfig,topologyContext);
     if(syncPolicy != null) {
       //if the user has specified the sync policy, we don't want to override their wishes.
+      LOG.debug("Using user specified sync policy {}", syncPolicy.getClass().getSimpleName());
       syncPolicyCreator = new ClonedSyncPolicyCreator(syncPolicy);
     }
     else {
       //if the user has not, then we want to have the sync policy depend on the batch size.
+      LOG.debug("No user specified sync policy, using CountSyncPolicy based on batch size");
       syncPolicyCreator = (source, config) -> new CountSyncPolicy(config == null?1:config.getBatchSize(source));
     }
   }
 
-
   @Override
   public BulkWriterResponse write(String sourceType
                    , WriterConfiguration configurations
                    , Iterable<Tuple> tuples
                    , List<JSONObject> messages
-                   ) throws Exception
-  {
+                   ) throws Exception {
     BulkWriterResponse response = new BulkWriterResponse();
 
     // Currently treating all the messages in a group for pass/failure.
-    try {
-      // Messages can all result in different HDFS paths, because of Stellar Expressions, so we'll need to iterate through
-      for(JSONObject message : messages) {
-        String path = getHdfsPathExtension(
-                sourceType,
-                (String)configurations.getSensorConfig(sourceType).getOrDefault(IndexingConfigurations.OUTPUT_PATH_FUNCTION_CONF, ""),
-                message
-        );
+    // Messages can all result in different HDFS paths, because of Stellar Expressions, so we'll need to iterate through
+    for (JSONObject message : messages) {
+      String path = getHdfsPathExtension(
+          sourceType,
+          (String) configurations.getSensorConfig(sourceType)
+              .getOrDefault(IndexingConfigurations.OUTPUT_PATH_FUNCTION_CONF, ""),
+          message
+      );
+
+      try {
+        LOG.trace("Writing message {} to path: {}", message.toJSONString(), path);
         SourceHandler handler = getSourceHandler(sourceType, path, configurations);
         handler.handle(message, sourceType, configurations, syncPolicyCreator);
+      } catch (Exception e) {
+        LOG.error(
```

Comment:
I don't think the `{}` substitution works with LOG.error.  I think you have to build the error string first like so.
```
String err = String.format("HdfsWriter encountered.. %s... blah blah.", sourceType, messages.size(), path);
LOG.error(err, e);
```

Reply:
As of slf4j 1.6.0 (we use 1.7.7), it works and the last arg has to be the exception (check out https://www.slf4j.org/faq.html#paramException). The example they give is

```
String s = "Hello world";
try {
  Integer i = Integer.valueOf(s);
} catch (NumberFormatException e) {
  logger.error("Failed to format {}", s, e);
}
```

Reply:
That's awesome. I've been doing it the hard way!